powerspikegg/computation_models/match/graph.py

```diff
@@ -25,7 +25,45 @@ class GraphBuilder:
     def __init__(self, input_size):
         self.input_size = input_size
 
-    def inference(self, data, hidden1_units, hidden2_units):
+    def add_hidden_layer(self, data, input_size, output_size, name,
+                         keep_probability, is_training):
+        with tf.name_scope(name):
+            weights = tf.Variable(
+                tf.truncated_normal([input_size, output_size],
+                                    stddev=0.1), name="weights")
+            tf.summary.histogram('weights', weights)
+            biases = tf.Variable(tf.zeros([output_size]), name="biases")
+            mult = tf.matmul(data, weights) + biases
+
+            # Activation
+            hidden1 = tf.nn.relu(mult)
+
+            # Batch normalization
+            norm = tf.contrib.layers.batch_norm(hidden1, 0.9,
+                                                is_training=is_training,
+                                                updates_collections=None)
+
+            # Droupout
+            dropped = tf.nn.dropout(norm, keep_probability)
+
+            return norm
+
+    def createNetwork(self, data, layers, keep_probability=1,
+                      is_training=True):
+        current = 1
+        hidden = tf.contrib.layers.batch_norm(data, 0.9,
```

Comment:
Just a side note FYI, you can deal with the `pep8` line-breaking and indentation level by doing this:

```
xxx = function(
    foo, bar, baz,
    foobar,...)
```